<launch>

	<!--DEPTH-->
	<node name="controller" pkg="pid" type="controller" ns="depth_pid" output="screen">
	    <param name="node_name" value="pid" />

	    <!--Default PID Gains (Should load from file eventually)-->
	    <param name="Kp" value="10" />
	    <param name="Ki" value="0.0" />
	    <param name="Kd" value="0.0" />

	    <!--Effort Limits-->
	    <param name="upper_limit" value="1000" />
	    <param name="lower_limit" value="-1000" />

	    <!--Integral Factor Limit-->
	    <param name="windup_limit" value="10" />

	    <!--Set Topics-->
	  	<!--Setpoint will be "/<ns>/setpoint"-->
	  	<!--Enable will be "/<ns>/pid_enable"-->
	  	<param name="topic_from_controller" value="/controls/superimposer/heave" />
	  	<param name="topic_from_plant" value="/state_estimation/depth" />
	</node>

    <!-- THESE ARE THE CV PIDS. We might want to migrate these to the CV package later -->
	<!--CV YAW -->
	<node name="controller" pkg="pid" type="controller" ns="CV_yaw_pid" output="screen">
	    <param name="node_name" value="pid" />

	    <!--Default PID Gains (Should load from file eventually)-->
	    <param name="Kp" value="10" />
	    <param name="Ki" value="0.0" />
	    <param name="Kd" value="0.0" />

	    <!--Effort Limits-->
	    <param name="upper_limit" value="1000" />
	    <param name="lower_limit" value="-1000" />

	    <!--Integral Factor Limit-->
	    <param name="windup_limit" value="10" />

	    <!--Set Topics-->
	  	<!--Setpoint will be "/<ns>/setpoint"-->
	  	<!--Enable will be "/<ns>/pid_enable"-->
	  	<param name="topic_from_controller" value="/controls/superimposer/yaw" />
	  	<param name="topic_from_plant" value="/cv/down_cam_heading_Fitting" />
	</node>
	
	<!--CV SWAY -->
	<node name="controller" pkg="pid" type="controller" ns="CV_sway_pid" output="screen">
	    <param name="node_name" value="pid" />

	    <!--Default PID Gains (Should load from file eventually)-->
	    <param name="Kp" value="10" />
	    <param name="Ki" value="0.0" />
	    <param name="Kd" value="0.0" />

	    <!--Effort Limits-->
	    <param name="upper_limit" value="1000" />
	    <param name="lower_limit" value="-1000" />

	    <!--Integral Factor Limit-->
	    <param name="windup_limit" value="10" />

	    <!--Set Topics-->
	  	<!--Setpoint will be "/<ns>/setpoint"-->
	  	<!--Enable will be "/<ns>/pid_enable"-->
	  	<param name="topic_from_controller" value="/controls/superimposer/sway" />
	  	
	  	<!--<param name="topic_from_plant" value="/cv/down_cam_target_centroid/gravity/x" /> -->
	  	<param name="topic_from_plant" value="/cv/down_cam_PIDx" />	  	
	  	
	</node>
	
	<!--CV SURGE -->
	<node name="controller" pkg="pid" type="controller" ns="CV_surge_pid" output="screen">
	    <param name="node_name" value="pid" />

	    <!--Default PID Gains (Should load from file eventually)-->
	    <param name="Kp" value="10" />
	    <param name="Ki" value="0.0" />
	    <param name="Kd" value="0.0" />

	    <!--Effort Limits-->
	    <param name="upper_limit" value="1000" />
	    <param name="lower_limit" value="-1000" />

	    <!--Integral Factor Limit-->
	    <param name="windup_limit" value="10" />

	    <!--Set Topics-->
	  	<!--Setpoint will be "/<ns>/setpoint"-->
	  	<!--Enable will be "/<ns>/pid_enable"-->
	  	<param name="topic_from_controller" value="/controls/superimposer/surge" />
	  	
	  	<!--<param name="topic_from_plant" value="/cv/down_cam_target_centroid/gravity/y" /> -->
	  	<param name="topic_from_plant" value="/cv/down_cam_PIDy" />	  
	</node>

	

	<!--SURGE-->
	<node name="controller" pkg="pid" type="controller" ns="surge_pid" output="screen">
	    <param name="node_name" value="pid" />

	    <!--Default PID Gains (Should load from file eventually)-->
	    <param name="Kp" value="10" />
	    <param name="Ki" value="0.0" />
	    <param name="Kd" value="0.0" />

	    <!--Effort Limits-->
	    <param name="upper_limit" value="1000" />
	    <param name="lower_limit" value="-1000" />

	    <!--Integral Factor Limit-->
	    <param name="windup_limit" value="10" />

	    <!--Set Topics-->
	  	<!--Setpoint will be "/<ns>/setpoint"-->
	  	<!--Enable will be "/<ns>/pid_enable"-->
	  	<param name="topic_from_controller" value="/controls/superimposer/surge" />
	  	<param name="topic_from_plant" value="/state_estimation/depth" />
	</node>



	<!--SWAY-->
	<node name="controller" pkg="pid" type="controller" ns="sway_pid" output="screen">
	    <param name="node_name" value="pid" />

	    <!--Default PID Gains (Should load from file eventually)-->
	    <param name="Kp" value="10" />
	    <param name="Ki" value="0.0" />
	    <param name="Kd" value="0.0" />

	    <!--Effort Limits-->
	    <param name="upper_limit" value="1000" />
	    <param name="lower_limit" value="-1000" />

	    <!--Integral Factor Limit-->
	    <param name="windup_limit" value="10" />

	    <!--Set Topics-->
	  	<!--Setpoint will be "/<ns>/setpoint"-->
	  	<!--Enable will be "/<ns>/pid_enable"-->
	  	<param name="topic_from_controller" value="/controls/superimposer/sway" />
	  	<param name="topic_from_plant" value="/state_estimation/depth" />
	</node>













	<!--YAW-->
	<node name="controller" pkg="pid" type="controller" ns="yaw_pid" output="screen">
	    <param name="node_name" value="pid" />

	    <!--IMPORTANT: This allows the pid to wrap values (i.e on [+pi,-pi])-->
	    <param name="angle_error" value="true" />

	    <!--Default PID Gains (Should load from file eventually)-->
	    <param name="Kp" value="5.0" />
	    <param name="Ki" value="0.5" />
	    <param name="Kd" value="0.1" />

	    <!--Effort Limits-->
	    <param name="upper_limit" value="10" />
	    <param name="lower_limit" value="-10" />

	    <!--Integral Factor Limit-->
	    <param name="windup_limit" value="10" />

	    <!--Set Topics-->
	  	<!--Setpoint will be "/<ns>/setpoint"-->
	  	<!--Enable will be "/<ns>/pid_enable"-->
	  	<param name="topic_from_controller" value="/controls/superimposer/yaw" />
	  	<param name="topic_from_plant" value="/state_estimation/yaw" />
	</node>

	<!--ROLL-->
	<node name="controller" pkg="pid" type="controller" ns="roll_pid" output="screen">
	    <param name="node_name" value="pid" />

	    <!--IMPORTANT: This allows the pid to wrap values (i.e on [+pi,-pi])-->
	    <param name="angle_error" value="true" />

	    <!--Default PID Gains (Should load from file eventually)-->
	    <param name="Kp" value="5.0" />
	    <param name="Ki" value="0.5" />
	    <param name="Kd" value="0.1" />

	    <!--Effort Limits-->
	    <param name="upper_limit" value="10" />
	    <param name="lower_limit" value="-10" />

	    <!--Integral Factor Limit-->
	    <param name="windup_limit" value="10" />

	    <!--Set Topics-->
	  	<!--Setpoint will be "/<ns>/setpoint"-->
	  	<!--Enable will be "/<ns>/pid_enable"-->
	  	<param name="topic_from_controller" value="/controls/superimposer/roll" />
	  	<param name="topic_from_plant" value="/state_estimation/roll" />
	</node>

	<!--PITCH-->
	<node name="controller" pkg="pid" type="controller" ns="pitch_pid" output="screen">
	    <param name="node_name" value="pid" />

	    <!--IMPORTANT: This allows the pid to wrap values (i.e on [+pi,-pi])-->
	    <param name="angle_error" value="true" />

	    <!--Default PID Gains (Should load from file eventually)-->
	    <param name="Kp" value="5.0" />
	    <param name="Ki" value="0.5" />
	    <param name="Kd" value="0.1" />

	    <!--Effort Limits-->
	    <param name="upper_limit" value="10" />
	    <param name="lower_limit" value="-10" />

	    <!--Integral Factor Limit-->
	    <param name="windup_limit" value="10" />

	    <!--Set Topics-->
	  	<!--Setpoint will be "/<ns>/setpoint"-->
	  	<!--Enable will be "/<ns>/pid_enable"-->
	  	<param name="topic_from_controller" value="/controls/superimposer/pitch" />
	  	<param name="topic_from_plant" value="/state_estimation/pitch" />
	</node>

</launch>
